hack/update-deps.sh

```diff
@@ -7,6 +7,56 @@ set -o pipefail
 cd "$(dirname "${BASH_SOURCE}")/.."
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+if [[ -z "${KUBE_BRANCH:-}" ]]; then
```

Comment:
I'm probably reading this wrong, but does this mean we can no longer run `hack/update-deps.sh` without setting this env var?

Reply:
Would like to base this off the pinned level in glide.yaml, this was just to make sure the approach worked